### DISCUSSION TITLE
Request had insufficient authentication scopes.

### DISCUSSION THREAD

**Discussion by erenkulaksiz** [imdhemy/laravel-in-app-purchases#394] [Q&A] [answered]
Hello,

I have an Android application with multiple subscriptions. I want to validate these receipts via my Laravel backend.

SubService.php:
```php
<?php

namespace App\Services;

use Illuminate\Support\Facades\Validator;
use App\Enums\ResponseMessageEnums;
use App\Enums\SubscriptionMessageEnums;
use App\Models\User;
use Illuminate\Http\Request;
use Imdhemy\GooglePlay\Subscriptions\SubscriptionPurchase;
use Imdhemy\Purchases\Facades\Subscription;

class SubService {
    public static function validateSubscription(Request $request) {
        $validator = Validator::make(request()->all(), [
            'itemId' => 'required|string',
            'purchaseToken' => 'required|string',
        ], [
            'itemId.required' => SubscriptionMessageEnums::INVALID_ITEM_ID,
            'itemId.string' => SubscriptionMessageEnums::INVALID_ITEM_ID,
            'purchaseToken.required' => SubscriptionMessageEnums::INVALID_PURCHASE_TOKEN,
            'purchaseToken.string' => SubscriptionMessageEnums::INVALID_PURCHASE_TOKEN,
        ]);

        if($validator->fails()) {
            return ['response' => ['message' => $validator->errors()->first()], 'status' => 400];
        }

        $itemId = $validator->validated()['itemId'];
        $purchaseToken = $validator->validated()['purchaseToken'];

        $subscriptionReceipt = Subscription::googlePlay()->id($itemId)->token($purchaseToken)->get();

        return ['response' => ['data' => 'test'], 'status' => 200];
    }
}
```

I have created api key on google cloud, with 100% permissions. 
But i get this error when i try to use this route:

`
GuzzleHttp\Exception\ClientException: Client error: `GET https://androidpublisher.googleapis.com/androidpublisher/v3/applications/[REDACTED_APP_ID]/purchases/subscriptions/[REDACTED_SUB_ID].yearly/[REDACTED_PURCHASE_TOKEN]` resulted in a `403 Forbidden` response:
{
  &quot;error&quot;: {
    &quot;code&quot;: 403,
    &quot;message&quot;: &quot;Request had insufficient authentication scopes.&quot;,
    &quot;errors&quot;: [
       (truncated...)
 in file vendor\guzzlehttp\guzzle\src\Exception\RequestException.php on line 113
`

What can cause this? I couldnt find any workarounds and i searched everywhere but i couldnt find any solution.

**Comment by imdhemy** [comment, marked as answer]
Salam @erenkulaksiz 
This issue is related to the Google credentials, not the library. Sometimes it takes a little bit of time to work as expected. This guide may help you fixing the scope issue. https://imdhemy.com/laravel-iap-docs/docs/credentials/google-play

**Reply by erenkulaksiz** [reply]
Already followed that documents like 5 times. No result. I used other library called laravel store receipt validator and it works fine now. Thanks for the reply though!
